### ISSUE TITLE
[java] SingularField false positive with non static inner classes

### ISSUE THREAD

**Issue by arturbosch** [pmd/pmd#1688]
**Affects PMD Version:** 6.11.0

**Rule:** SingularField

**Description:** Suggests to move the field `value` to local var even if it is used from an inner class instance with `A.this.value`.

**Code Sample demonstrating the issue:**

```java
class A {

    private String value;
    private String derivedValue;

    public A(String value) {
        this.value = value;
        this.derivedValue = this.value + "-more";
    }

    private void run() {
        final B b = new B();
        b.useA();
    }

    public static void main(String[] args) {
        new A("").run();
    }

    public class B {

        public void useA() {
            final String field = A.this.value;
            System.out.println(field);
        }
    }
}

```

**Running PMD through:** CLI
